clang/lib/Sema/SemaOverload.cpp

```diff
@@ -12114,6 +12115,35 @@ static void NoteFunctionCandidate(Sema &S, OverloadCandidate *Cand,
     return;
   }
 
+  // If this is an implicit deduction guide against a non-user-defined
+  // constructor, add a note for it. These deduction guides nor their
+  // corresponding constructors are not explicitly spelled in the source code,
+  // and simply producing a deduction failure note around the heading of the
+  // enclosing RecordDecl is confusing.
+  //
+  // We prefer adding such notes at the end of the last deduction failure
+  // reason because duplicate code snippets appearing in the diagnostic
+  // are likely becoming noisy.
```

Comment:
```suggestion
  // If this is an implicit deduction guide against an implicitly defined
  // constructor, add a note for it. Neither these deduction guides nor their
  // corresponding constructors are explicitly spelled in the source code,
  // and simply producing a deduction failure note around the heading of the
  // enclosing RecordDecl would be confusing.
  //
  // We prefer adding such notes at the end of the last deduction failure
  // reason because duplicate code snippets appearing in the diagnostic
  // would likely become noisy.
```
Just a few minor changes here too